src/main/java/cz/cvut/kbss/termit/persistence/dao/TermDao.java

```diff
@@ -387,10 +387,25 @@ private String r(String string, String from, String to) {
     private List<TermDto> loadIncludedTerms(Collection<URI> includeTerms) {
         return includeTerms.stream().map(u -> em.find(TermDto.class, u))
                            .filter(Objects::nonNull)
-                           .peek(t -> t.setSubTerms(loadSubTerms(t)))
+                           .peek(this::recursivelyLoadParentTermSubTerms)
                            .collect(Collectors.toList());
     }
 
+    /**
+     * Recursively loads subterms for the specified term and its parents (if they exist).
+     * <p>
+     * This implementation ensures that the term hierarchy can be traversed both ways for the specified term. This has
+     * to be done to allow the tree-select component on the frontend to work properly and display the terms.
+     *
+     * @param term The term to load subterms for
+     */
+    private void recursivelyLoadParentTermSubTerms(TermDto term) {
+        term.setSubTerms(loadSubTerms(term));
+        if (term.hasParentTerms()) {
+            term.getParentTerms().forEach(this::recursivelyLoadParentTermSubTerms);
```

Comment:
I expect this is efficient enough, as - if traversed bottom-up - children of parents might be found in the cache.